dolphinscheduler-ui/src/views/projects/task/components/node/fields/use-flink.ts

```diff
@@ -36,14 +36,51 @@ export function useFlink(model: { [field: string]: any }): IJsonItem[] {
   )
 
   const taskManagerNumberSpan = computed(() =>
-    model.flinkVersion === '<1.10' && model.deployMode === 'cluster' ? 12 : 0
+    model.flinkVersion === '<1.10' && model.deployMode !== 'local' ? 12 : 0
   )
 
   const deployModeSpan = computed(() =>
-    model.deployMode === 'cluster' ? 12 : 0
+    model.deployMode !== 'local' ? 12 : 0
   )
 
-  const appNameSpan = computed(() => (model.deployMode === 'cluster' ? 24 : 0))
+  const appNameSpan = computed(() => (model.deployMode !== 'local' ? 24 : 0))
+
+  const deployModeOptions = computed(() => {
+    if (model.flinkVersion === '<1.10') {
+      return [
+        {
+          label: 'cluster',
+          value: 'cluster'
+        },
+        {
+          label: 'local',
+          value: 'local'
+        }
+      ];
+    } else {
+      return [
+        {
+          label: 'per-job/cluster',
+          value: 'cluster'
+        },
+        {
+          label: 'application',
+          value: 'application'
+        },
+        {
+          label: 'local',
+          value: 'local'
+        }
+      ];
+    }
+  })
+
+  watch(
+    () => model.flinkVersion,
+    () => {
+      model.deployMode = 'cluster'
+    }
+  )
```

Comment:
There will be a bug here after setting deployMode to application and flinkVersion to 1.11.

Reply:
Hi @Amy0104 ,
I updated use-flink.ts. Now deploy mode will reset to 'cluster' only when flink version is '<1.10' and deploy mode is 'application'.